### ISSUE TITLE
Error: TTS not ready in Android

### ISSUE THREAD

**Issue by bhakta0007** [ak1394/react-native-tts#66]
Works on IOS emulator but get the "Error: TTS not ready" on Android (tried both simulator as well as device with same result).

I have verified that TTS works when tested under settings on emulator as well as device. 

Android emulator version: 8.0.0
Device version: 7.1.2

the error in console log:

```
blob:http://localhost:8081/463ce249-4719-4e5f-ad7d-a2f63dc584b8:sourcemap:53509 Possible Unhandled Promise Rejection (id: 0):
Error: TTS is not ready
Error: TTS is not ready
    at createErrorFromErrorData (blob:http://localhost:8081/463ce249-4719-4e5f-ad7d-a2f63dc584b8:2023:17)
    at blob:http://localhost:8081/463ce249-4719-4e5f-ad7d-a2f63dc584b8:1975:27
    at MessageQueue.__invokeCallback (blob:http://localhost:8081/463ce249-4719-4e5f-ad7d-a2f63dc584b8:2417:18)
    at blob:http://localhost:8081/463ce249-4719-4e5f-ad7d-a2f63dc584b8:2162:18
    at MessageQueue.__guardSafe (blob:http://localhost:8081/463ce249-4719-4e5f-ad7d-a2f63dc584b8:2330:11)
    at MessageQueue.invokeCallbackAndReturnFlushedQueue (blob:http://localhost:8081/463ce249-4719-4e5f-ad7d-a2f63dc584b8:2161:14)
    at http://localhost:8081/debugger-ui/debuggerWorker.js:72:58
```

**Comment by ak1394** [comment]
Have you tried waiting for init to finish as in
`Tts.getInitStatus().then(() => {
  Tts.speak('Hello, world!');
});`
?

**Comment by bhakta0007** [comment]
Yes I did that, it never got into the then part. I have placed the code inside a onClick of a button

**Comment by ak1394** [comment]
It looks like it's something specific to the device you're trying it on. Perhaps some assets have to be downloaded for TTS to work, or something like that.

**Comment by fungilation** [comment]
@ak1394 this issue shouldn't be closed as it's a show stopper on Android. I've been using this great lib on iOS without issue, until I'm now developing my app for Android and now keep hitting this issue with exact same console error (yellow box) output.

I'm testing on virgin install of Nexus 5 - Android 6.0.1. Even if users need to install some voice pack first, it's not really acceptable to have app bugging out with a yellow box or crashing out without better fault tolerance in catching the error.

EDIT: after investigating, I find it's reliance on Tts.getInitStatus() to run init function calls that's the problem. After my android device rebooted and having wrapped those calls in Tts.getInitStatus() block (new to me, I didn't use it before), it would *always* fail on Android with error output on first comment of this issue. I tried everything and it won't stop failing. Catch the error and it stops the yellow box but it results in the same error, and it would fail and caught in a .catch() at about precisely 2 minutes after Tts.getInitStatus() was ran.

So I tried taking out Tts.getInitStatus() and surprise, tts works again on Android! These are the init code I run, inside the Tts.getInitStatus().then() block (or not, rather):

```javascript
Tts.setDefaultLanguage('en-US')
Tts.setDefaultRate(0.54)   // 0.01 is a slowest rate and 0.99 is the fastest rate. Was first set to 0.55
Tts.addEventListener('tts-start', (event) => {})    // console.log("Tts start", event)
Tts.addEventListener('tts-cancel', (event) => {})   // console.log("Tts cancel", event)
Tts.addEventListener('tts-finish', (event) => {
...
(app logic, including looping through data and continuing with more Tts.speak() calls)
}
```

And it seems that once Tts.setDefaultLanguage or the rest are run just once, I can't reproduce Tts.getInitStatus() erroring out again. I can't explain this behaviour on android, and I don't do any native android dev. Hopefully my observations are useful for this weird behaviour. But for my app, I'm better off not relying on Tts.getInitStatus() at all, but simply run my tts init code inside a setTimeout with 2s delay inside a component constructor.
